### ISSUE TITLE
Crash when linting template literal inside an object declaration

### ISSUE THREAD

**Issue by JeremyPouyet** [eslint/eslint#14022]
**Tell us about your environment**

Node version: v14.15.0
npm version: v6.14.11
Local ESLint version: Not found
Global ESLint version: v7.18.0 (Currently used)

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

babel-eslint (I do not use babel, I just use the linter)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    es6: true,
    node: true,
    mocha: true
  },
  globals: { logger: true, IO: true, NLP: true },
  extends: 'eslint:recommended',
  parser: 'babel-eslint',
  parserOptions: {
    sourceType: 'script',
    ecmaVersion: 12
  },
  rules: {
    'arrow-spacing': 'error',
    'comma-dangle': [ 'error', 'never' ],
    'comma-spacing': [ 'error', {
      before: false,
      after: true
    }],
    complexity: [ 'warn', 15 ],
    'computed-property-spacing': [ 'error', 'never' ],
    'consistent-return': 'warn',
    indent: [ 'error', 2, {
      SwitchCase: 1
    }],
    'dot-notation': 'warn',
    'eol-last': [ 'error', 'always' ],
    'func-call-spacing': [ 'error', 'never' ],
    'key-spacing': [ 'error', {
      mode: 'minimum',
      beforeColon: false,
      afterColon: true
    }],
    'linebreak-style': [ 'error', 'unix' ],
    'max-len': [ 'warn', 100 ],
    'max-lines': [ 'warn', {
      max: 250,
      skipBlankLines: true,
      skipComments: true
    }],
    'max-params': [ 'warn', 4 ],
    'max-statements': [ 'warn', 30 ],
    'new-parens': 'error',
    'no-compare-neg-zero': 'warn',
    'no-else-return': 'warn',
    'no-empty': [ 'error', {
      allowEmptyCatch: true
    }],
    'no-floating-decimal': 'error',
    'no-labels': 'error',
    'no-lonely-if': 'error',
    'no-multiple-empty-lines': [ 'error', {
      max: 2,
      maxBOF: 0,
      maxEOF: 1
    }],
    'no-multi-spaces': 'off',
    'no-multi-str': 'error',
    'no-nested-ternary': 'error',
    'no-unused-vars': [ 'error' ],
    'no-useless-concat': 'error',
    'object-curly-spacing': [ 'error', 'always' ],
    'object-curly-newline': [ 'error', {
      multiline: true,
      consistent: true
    }],
    'object-property-newline': [ 'error', {
      allowMultiplePropertiesPerLine: true
    }],
    'operator-linebreak': [ 'error', 'before' ],
    'prefer-promise-reject-errors': 'error',
    'prefer-const': 'warn',
    radix: [ 'error', 'always' ],
    semi: [ 'error', 'always' ],
    'semi-style': [ 'error', 'last' ],
    'semi-spacing': [ 'error', {
      before: false,
      after: true
    }],
    'space-before-function-paren': [ 'error', 'never' ],
    'spaced-comment': [ 'error', 'always' ],
    'space-unary-ops': [ 'error', {
      words: false,
      nonwords: false
    }],
    strict: [ 'error', 'global' ],
    quotes: [ 'warn', 'single', {
      allowTemplateLiterals: true
    }],
    'valid-jsdoc': [ 'warn', {
      'requireReturn': false
    }],
    'wrap-iife': [ 'error', 'inside' ]
  };
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const minutes = 5;
const maxRequest = 200;

// const limiter = `Too many requests. ${maxRequest} per ${minutes} allowed.` is working
const limiter = {
  message: `Too many requests. ${maxRequest} per ${minutes} allowed.`
};
```

```bash
eslint file.js
```

**What actually happened? Please include the actual, raw output from ESLint.**

Oops! Something went wrong! :(

ESLint: 7.18.0

```
TypeError: Cannot read property 'range' of null
Occurred while linting /home/jeremy/project/packages/api/file.js:1
    at OffsetStorage.setDesiredOffset (/home/jeremy/.nvm/versions/node/v14.15.0/lib/node_modules/eslint/lib/rules/indent.js:341:45)
    at /home/jeremy/.nvm/versions/node/v14.15.0/lib/node_modules/eslint/lib/rules/indent.js:1396:29
    at Array.forEach (<anonymous>)
    at Object.TemplateLiteral [as listener] (/home/jeremy/.nvm/versions/node/v14.15.0/lib/node_modules/eslint/lib/rules/indent.js:1388:34)
    at /home/jeremy/.nvm/versions/node/v14.15.0/lib/node_modules/eslint/lib/rules/indent.js:1634:55
    at Array.forEach (<anonymous>)
    at Program:exit (/home/jeremy/.nvm/versions/node/v14.15.0/lib/node_modules/eslint/lib/rules/indent.js:1634:26)
    at /home/jeremy/.nvm/versions/node/v14.15.0/lib/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/jeremy/.nvm/versions/node/v14.15.0/lib/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
```

**Are you willing to submit a pull request to fix this bug?**
Of course

**Comment by aladdin-add** [comment]
does is happen using the default parser (by removing `parser: 'babel-eslint',` in your eslintrc)?

**Comment by JeremyPouyet** [comment]
The issue does not happen when I remove `parser: 'babel-eslint'` from my config. However I did not update babel-eslint recently.
I've already tested with @babel/eslint-parser before creating this issue but the same crash happened.

**Comment by aladdin-add** [comment]
so it seems related to the parser, maybe you can fill the issue in babel repo.  :)

**Comment by JeremyPouyet** [comment]
I'll open an issue on babel repo but the bug appeared after I've updated the eslint package whereas the version of babel-eslint I used has more than a year.

**Comment by ChrisWiles** [comment]
Mine just started crashing after updating some dependencies, I am not using babel-eslint. 
ESLint: 7.18.0

Crashing when running `node_modules/eslint/lib/rules/template-curly-spacing.js`

Still digging into the exact issue.

**Comment by furkanmustafa** [comment]
Had the same issue. Upstream issue here https://github.com/babel/babel-eslint/issues/799.
Some of those solutions worked for me (still using latest versions).

**Comment by mdjermanovic** [comment]
Hi @furkanmustafa!

That's an issue about `babel-eslint`, which has been deprecated in favor of `@babel/eslint-parser`. Can you try with the latest version of  [@babel/eslint-parser](https://www.npmjs.com/package/@babel/eslint-parser)?

**Comment by furkanmustafa** [comment]
I am already using latest version (7.12.13) of `@babel/eslint-parser`. Apparently they still manage eslint integration issues on the old repository for `eslint-parser` (or at the time of that issue they were using that repository?). And as I mentioned, my setup is now working after applying suggestions on the issue.

My package.json currently looks like this;

```
  ...
  "devDependencies": {
    "@babel/core": "^7.12.13",
    "@babel/eslint-parser": "^7.12.13",
    "@babel/parser": "^7.12.15",
    "@babel/plugin-proposal-class-properties": "^7.12.13",
    "@babel/preset-env": "^7.12.13",
    "@babel/traverse": "^7.12.13",
    "@babel/types": "^7.12.13",
    "eslint": "^7.19.0",
    "eslint-plugin-vue": "^7.3.0",
    ....
  },
  ...
```

I think adding `@babel/parser`, `@babel/traverse` and `@babel/types` to package.json fixed it. I didn't try to figure out which specific one was the cause, very high likely `@babel/parser`.

**Comment by mdjermanovic** [comment]
[babel/babel-eslint](https://github.com/babel/babel-eslint) repository has been archived, so if anyone is still having the same problem with the latest `@babel/eslint-parser` then it might be good to open a new issue on [babel/babel](https://github.com/babel/babel).

> I think adding `@babel/parser`, `@babel/traverse` and `@babel/types` to package.json fixed it. I didn't try to figure out which specific one was the cause, very high likely `@babel/parser`.

It's quite possible. I believe the root cause of the problem has been fixed several months ago in the package where it originated, but it could be that some dependencies between babel packages are not updated to enforce the use of that version with the latest `@babel/eslint-parser`. It could be also that the dependencies are okay, but package managers sometimes miss to update everything that should be updated in existing `node_modules`.

**Comment by nzakas** [comment]
Closing, as this is an issue with the Babel parser, which we don’t maintain.
